src/qibo/models/circuit.py

```diff
@@ -1046,21 +1039,27 @@ def compile(self, backend=None):
 
         if self.compiled:
             raise_error(RuntimeError, "Circuit is already compiled.")
+
         if not self.queue:
             raise_error(RuntimeError, "Cannot compile circuit without gates.")
+
         for gate in self.queue:
             if isinstance(gate, gates.CallbackGate):  # pragma: no cover
                 raise_error(
                     NotImplementedError,
                     "Circuit compilation is not available with callbacks.",
                 )
 
-        from qibo.backends import _check_backend
+        from qibo.backends import (  # pylint: disable=import-outside-toplevel
+            _check_backend,
+        )
+        from qibo.result import (  # pylint: disable=import-outside-toplevel
+            CircuitResult,
+            QuantumState,
+        )
```

Comment:
If not possible top-level, at least let's put these at the beginning of the method (I know, they were already there, and you already regrouped them, which is a first step).